### ISSUE TITLE
feat: improving command search

### ISSUE THREAD

**Issue by raydreams** [avalynndev/enjoytown#129]
0-0

**Comment by heyitsleo** [comment]
Another place for ideas relating to search, and other things such as layouts:
https://movies-topaz-gamma.vercel.app | https://github.com/oktay/movies

**Comment by avalynndev** [comment]
> Another place for ideas relating to search, and other things such as layouts: https://movies-topaz-gamma.vercel.app | https://github.com/oktay/movies

Hey I explored the search bar and prefer to stay with my current search bar. On the other hand the layout looks way better, I'l try to improve my website with hints from that website.

**Comment by avalynndev** [comment]
Here's the points I inferred from your issue a long with a valid reply to each one of them:-
- automatically sort by relevance or popularity: that's a pretty good idea. I will implement it soon
- user's past searches: also a good idea, will add it
 - user/ip based watchlists: I don't like the idea of fetching the IP address of anyone, and watchlists in an upcoming feature in the project, planned way before. However the release date of this watchlists feat is way longer than you think, as we have to add a proper login, signup a long with lots of other features
- add recommendations/similar shows or movies: this is already in progress will be updated once I finish my exams
- adding filters at the main movie/show browsing page, or in the search menu: This is a good idea, I will try to add it (might take time cuz I have no idea how to make it)

in your comment I haven't understood this sentence clearly, mind explaining it to me more descriptively?
> selected episodes/seasons when they search and re-open the page again because some players save the watch progress.

**Comment by heyitsleo** [comment]
For my comment, I was saying to save the users watch history so when they re-open the page, they can resume watching it from where they left it off at. But, I just realized that it really wouldn't work without an account or saving on ip address so it'll have to wait.

**Comment by avalynndev** [comment]
> For my comment, I was saying to save the users watch history so when they re-open the page, they can resume watching it from where they left it off at. But, I just realized that it really wouldn't work without an account or saving on ip address so it'll have to wait.

Actually, that's a nice idea.. I'm making a account system so I can do that.. however you cant really go back to where u stopped ur movie as I'm using Iframe tags for tv shows and movies.. is that fine?

**Comment by heyitsleo** [comment]
Yes, I think it would be useful while you're also implementing recent searches. Some players do save your time so theres no need to worry on that.

**Comment by avalynndev** [comment]
Yep thats true

**Comment by avalynndev** [comment]
It aint late for anything my guy.. anywyas good idea didnt know i could do that lol

**Comment by heyitsleo** [comment]
I feel like for sorting, you can just remove the genres and use TMDB to fetch the results, then use plotwist's search card feature, (i linked the code aswell) https://developer.themoviedb.org/reference/search-multi

**Comment by heyitsleo** [comment]
^ Just to add on, filter is also included in oktay/movies. [movies-topaz-gamma.vercel.app](https://movies-topaz-gamma.vercel.app/) | [oktay/movies](https://github.com/oktay/movies)

**Comment by heyitsleo** [comment]
Sorry to add more work, but again if you want help with the command menu improvements:
I recommend adding the cards and using https://github.com/Vette1123/movies-streaming-platform/blob/main/components/command-menu.tsx as some help.

**Comment by avalynndev** [comment]
0-0
@GitGitro.. any help with this my guy?

**Comment by GitGitro** [comment]
I'll see what i can do, i just started learning nextjs and typescript after all sorry 😅

**Comment by avalynndev** [comment]
No problem mg take your time 0-0.. better late than never

**Comment by heyitsleo** [comment]
Yeah, like I've done some small things as I'm not to advanced such as adding icons to the dropdown menu in the navbar, or revolving text. Take your tine guys!

**Comment by avalynndev** [comment]
Note: i am deleting comments to make it clean so i can work on it 0-0

**Comment by avalynndev** [comment]
https://github.com/users/avalynndev/projects/1/views/1
added some items to upcoming roadmap

**Comment by avalynndev** [comment]
Fixed
